model/src/util.py

```diff
@@ -174,3 +176,31 @@ def tracked_value(self) -> torch.Tensor:
             raise ValueError("No data has been received yet")
 
         return self.variance
+
+
+class InhibitoryPlasticityTrace:
+
+    def __init__(self, tau_stdp: float = TAU_STDP) -> None:
+        """
+        Zenke's paper uses a tau_stdp of 20ms.
+
+        TODO: Concern here is that we are initializing the state with zeros,
+                which might not be the best approach.
+        """
+        self.trace: Optional[torch.Tensor] = None
+        self.tau_stdp: float = tau_stdp
+
+    def apply(self, spike: torch.Tensor, dt: float = DT) -> torch.Tensor:
+        if self.trace is None:
+            # Initialize variance based on the first spike received
+            self.trace = torch.zeros_like(spike)
+
+        self.trace += (-1) * (dt / self.tau_stdp) * self.trace + spike
```

Comment:
I want to call out that this is a different form than the others for moving averages that decay. Example where `dt/tau` is applied to the whole term:


https://github.com/and-rewsmith/LPL-SNN/blob/2a1466b4f5fb8bb4c76d2b7b03b9a417b3360289/model/src/util.py#L134

Reply:
Agreed, I believe that this manifests itself as a tau/dt scale factor on the STDP rule. Do you see any other ramifications of this choice?

Reply:
> Agreed, I believe that this manifests itself as a tau/dt scale factor on the STDP rule. 

I have added this to the list of things to discuss next time we chat. Will need to understand this more.